src/features/authentication/authenticationSlice.ts

```diff
@@ -0,0 +1,136 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import {
+  AuthStoreType,
+  AuthenticationSteps,
+} from "../../lib/authentication/authTypes";
+import {
+  Auth0AuthProvider,
+  authConfig,
+} from "../../lib/authentication/auth0AuthProvider";
+import {
+  MsalAuthProvider,
+  staticMsalConfig,
+  scopes,
+} from "../../lib/authentication/msalAuthProvider";
+import { m } from "framer-motion";
+
+//export const authProvider = new Auth0AuthProvider(authConfig);
```

Comment:
this "choice" should be in the App, not in the 'library' part.

Reply:
//export const authProvider = new Auth0AuthProvider(authConfig);
export const authProvider = new MsalAuthProvider(staticMsalConfig, scopes);

This line is the only one you have to change to switch authentication provider.
the difference in properties it is again to walk around the Env missing.

Do you mean that we have to switch in auth provider in app? like with a button?

Reply:
No, it's right to be a 'startup' thing, not a button.

But this choice (which implementation to istantiate) should be in the Main/App, not in the 'library'.

This slice should 'receive' the Provider from elsewhere.

Reply:
ok got it